src/shims/unix/sync.rs

```diff
@@ -80,10 +85,10 @@ fn mutex_reset_id<'mir, 'tcx: 'mir>(
 ) -> InterpResult<'tcx, ()> {
     ecx.deref_pointer_and_write(
         mutex_op,
-        4,
+        PTHREAD_MUTEX_ID,
         Scalar::from_i32(0),
         ecx.libc_ty_layout("pthread_mutex_t"),
-        ecx.machine.layouts.u32,
+        ecx.machine.layouts.i32,
```

Comment:
No this should be u32, we interpret this as an unsigned number. It doesn't really matter what the libc type does here.

Reply:
sorry I had a crash without this change but then now it works ... will roll back